SCClassLibrary/Common/Core/Object.sc

```diff
@@ -688,7 +688,7 @@ Object  {
 		StartUp.defer { // make sure the synth defs are written to the right path
 			var file;
 			dir = dir ? SynthDef.synthDefDir;
-			if (name.isNil) { Error("missing SynthDef file name").throw } {
+			if (name.isNil || name.asString.isEmpty) { Error("missing SynthDef file name").throw } {
```

Comment:
it doesn't matter here but you should know currently it's slightly more performant to write `||  xyz` as `or: { xyz }`

Reply:
I didn't know that, thanks! Is this documented?

Reply:
i don't know, @jamshark70 is the one who always stresses this